in_toto/models/metadata.py

```diff
@@ -23,20 +23,57 @@
 """
 
 import attr
+import copy
 import json
+from typing import Optional, List, Union
 
 import securesystemslib.keys
 import securesystemslib.formats
 import securesystemslib.exceptions
 import securesystemslib.gpg.functions
+from securesystemslib.key import Key
+from securesystemslib.metadata import Envelope as SSlibEnvelope
+from securesystemslib.signer import GPGSignature, Signature, Signer
+from securesystemslib.serialization import (BaseDeserializer, BaseSerializer,
+    JSONSerializer, SerializationMixin)
 
 from in_toto.models.common import ValidationMixin
 from in_toto.models.link import Link
 from in_toto.models.layout import Layout
-from in_toto.exceptions import SignatureVerificationError
+
+
+ENVELOPE_PAYLOAD_TYPE = "application/vnd.in-toto+json"
+
+
+# pylint: disable=import-outside-toplevel
+class AnyMetadata(SerializationMixin):
+  """A Metadata abstraction between DSSE Envelope and Metablock."""
```

Comment:
Should we mention how `AnyMetadata` can be used, either in words, or even with a little code snippet, e.g. as outlined here: https://github.com/theupdateframework/python-tuf/issues/2093#issuecomment-1234034698 ?

I think it would also be useful if we defined the methods that Envelope and Metablock have in common, both for future mypy runs and readthedocs.

Reply:
I still think this would be useful, but we can ticketize it.